### PR TITLE
tiflash late materialization

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-use-late-materialization.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+title: TiFlash 延迟物化
```
Comment (qiancai) [inline develop/dev-guide-use-late-materialization.md:2]: 其他 TiFlash 相关的文档大部分在参考指南这里，请看看这篇文档是否需要也放到这里
image

Comment (Lloyd-Pottiger) [reply develop/dev-guide-use-late-materialization.md:2]: no

File: develop/dev-guide-use-late-materialization.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# 延迟物化
+
+> **警告：**
+>
+> 该功能目前是实验性功能，其形式和使用方法可能会在未来版本中发生变化。
+
+本文档介绍通过使用 TiFlash 延迟物化来加速 Online Analytical Processing (OLAP) 场景中查询的方法。
+
+默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件进行过滤，聚合等计算任务。延迟物化是一种优化方式，它支持将部分过滤条件下推。先扫描过滤条件相关列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
+
```
Comment (qiancai) [inline develop/dev-guide-use-late-materialization.md:15]: 这里可以在文档给一个具体的 SQL 语句的例子说明吗？比如 
SELECT * FROM employee WHERE id IN (1, 2, 3) and age IN (10, 11, 12) 之类的

Comment (Lloyd-Pottiger) [reply develop/dev-guide-use-late-materialization.md:15]: done

**Review by flowbehappy — APPROVED** [review]
LGTM with a comment

File: develop/dev-guide-use-late-materialization.md (hunk 1 above)
Comment (Lloyd-Pottiger) [reply develop/dev-guide-use-late-materialization.md]: done

File: tiflash/tiflash-late-materialization.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+## 实现机制
+
+当有过滤条件下推到 TableScan 算子时，TableScan 算子过程整体包括了以下步骤：
+
+1. 读 <handle, del_mark, version> 三列，执行 MVCC 过滤，生成 MVCC Bitmap。
+2. 读过滤条件相关的列，执行过滤条件，生成 Filter Bitmap。
+3. 通过 MVCC Bitmap 和 Filter Bitmap 做与操作生成 Final Bitmap。
+4. 根据 Final Bitmap 读取剩余列的对应行。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:90]:
```suggestion
4. 根据 Final Bitmap 读取符合查询条件的列对应的行。
```

File: tiflash/tiflash-late-materialization.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+可以用下面语句禁用 TiFlash 延迟物化:
+
+```sql
+set session tidb_enable_late_materialization=OFF;
+set global tidb_enable_late_materialization=OFF;
+```
+
+## 实现机制
+
+当有过滤条件下推到 TableScan 算子时，TableScan 算子过程整体包括了以下步骤：
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:85]:
```suggestion
当有过滤条件下推到 TableScan 算子时，TableScan 算子的执行过程主要包括了以下步骤：
```

File: tiflash/tiflash-late-materialization.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
+---
+title: TiFlash 延迟物化
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:2]: 需要更新 TOC.md，把这篇文档  tiflash/tiflash-late-materialization.md 加进入，文档站左侧目录才会显示。

File: system-variables.md [hunk 6]
```diff
@@ -1379,6 +1379,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 表示是否启用 `IndexMergeJoin` 算子。
 - 该变量为 TiDB 内部变量，**不推荐使用**，否则可能会造成数据正确性问题。
 
+### `tidb_enable_late_materialization` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 表示是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能
```
Comment (qiancai) [inline system-variables.md:1388]:
```suggestion
- 表示是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。
```

File: system-variables.md [hunk 7]
```diff
@@ -1379,6 +1379,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 如果开启 TiFlash 延迟物化功能（设置为 ON 时）, 当 SELECT 语句中包含过滤条件（ WHERE 子句）时，普通的处理方式是扫描所有数据后进行过滤。TiFlash 延迟物化功能可以先扫描过滤条件相关列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```
Comment (qiancai) [inline system-variables.md:1389]:
```suggestion
- 如果开启 TiFlash 延迟物化功能（设置为 `ON` 时）, 当 `SELECT` 语句中包含过滤条件（ `WHERE` 子句）时，普通的处理方式是扫描查询所需列的全部数据后进行过滤。TiFlash 延迟物化功能可以先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```

File: tiflash/tiflash-late-materialization.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
+
+## 启用和禁用 TiFlash 延迟物化
+
+默认情况下，session 和 global 级别的变量 `tidb_enable_late_materialization=OFF`，即没有开启 TiFlash 延迟物化 功能。你可以通过以下语句来查看对应的变量信息。
+
+```
+show variables like 'tidb_enable_late_materialization';
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:41]:
```suggestion
SHOW VARIABLES LIKE 'tidb_enable_late_materialization';
```

File: tiflash/tiflash-late-materialization.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+```
++----------------------------------+-------+
+| Variable_name                    | Value |
++----------------------------------+-------+
+| tidb_enable_late_materialization | OFF   |
++----------------------------------+-------+
+```
+
+```
+show global variables like 'tidb_enable_late_materialization';
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:53]:
```suggestion
SHOW GLOBAL VARIABLES LIKE 'tidb_enable_late_materialization';
```

File: tiflash/tiflash-late-materialization.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+# 延迟物化
+
+> **警告：**
+>
+> 该功能目前是实验性功能，其形式和使用方法可能会在未来版本中发生变化。
+
+本文档介绍通过使用 TiFlash 延迟物化来加速 Online Analytical Processing (OLAP) 场景中查询的方法。
+
+默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件进行过滤，聚合等计算任务。延迟物化是一种优化方式，它支持将部分过滤条件下推。先扫描过滤条件相关列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tiflash/tiflash-late-materialization.md]: `它支持先下推部分过滤条件` 不准确，改成了 `它支持下推部分过滤条件 到 TableScan 算子`

File: tiflash/tiflash-late-materialization.md (hunk 3 above)
Comment (Lloyd-Pottiger) [reply tiflash/tiflash-late-materialization.md]: `剩余` 为了突出之前读过的列不会重复读

File: tiflash/tiflash-late-materialization.md (hunk 5 above)
Comment (Lloyd-Pottiger) [reply tiflash/tiflash-late-materialization.md:2]: done

File: tiflash/tiflash-late-materialization.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+```
++--------------------------------------+-------+
+| Variable_name                        | Value |
++--------------------------------------+-------+
+| tidb_opt_enable_late_materialization | OFF   |
++--------------------------------------+-------+
+```
+
+变量 `tidb_opt_enable_late_materialization` 支持 session 级别和 global 级别的修改。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:66]:
```suggestion
变量 `tidb_opt_enable_late_materialization` 支持 session 级别和 global 级别的修改。

```

File: tiflash/tiflash-late-materialization.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+# 延迟物化
+
+> **警告：**
+>
+> 该功能目前是实验性功能，其形式和使用方法可能会在未来版本中发生变化。
+
+本文档介绍通过使用 TiFlash 延迟物化的方式来加速 Online Analytical Processing (OLAP) 场景的查询。
+
+默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件对数据进行过滤、聚合等计算任务。延迟物化是一种优化方式，它支持下推部分过滤条件 到 TableScan 算子，即先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:14]:
```suggestion
默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件对数据进行过滤、聚合等计算任务。延迟物化是一种优化方式，它支持下推部分过滤条件到 TableScan 算子，即先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```

File: tiflash/tiflash-late-materialization.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+如果希望提升 OLAP 场景部分查询的性能，可以在 session 级别或 global 级别开启 TiFlash 延迟物化功能，你可以通过修改变量 `tidb_opt_enable_late_materialization` 的值来选择是否启用 TiFlash 延迟物化功能。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:16]:
```suggestion
如果希望提升 OLAP 场景部分查询的性能，可以在 session 级别或 global 级别开启 TiFlash 延迟物化功能。你可以通过修改变量 `tidb_opt_enable_late_materialization` 的值来选择是否启用 TiFlash 延迟物化功能。
```

File: tiflash/tiflash-late-materialization.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+启用 TiFlash 延迟物化功能后，TiDB 优化器会根据统计信息和查询的过滤条件，决定哪些过滤条件会被下推。优化器会优化考虑下推过滤率高的过滤条件，详细算法可以参考 [RFC 文档]()。
+
+例如：
+
+```
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:22]:
```suggestion
```sql
```

File: tiflash/tiflash-late-materialization.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+```
+
+过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
+
+## 启用和禁用 TiFlash 延迟物化
+
+默认情况下，session 和 global 级别的变量 `tidb_opt_enable_late_materialization=OFF`，即没有开启 TiFlash 延迟物化功能。你可以通过以下语句来查看对应的变量信息。
+
+```
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:42]:
```suggestion
```sql
```

File: tiflash/tiflash-late-materialization.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
++--------------------------------------+-------+
+| Variable_name                        | Value |
++--------------------------------------+-------+
+| tidb_opt_enable_late_materialization | OFF   |
++--------------------------------------+-------+
+```
+
+```
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:54]:
```suggestion
```sql
```

File: tiflash/tiflash-late-materialization.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+| id                      | estRows  | task         | access object | operator info                                         |
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+| TableReader_12          | 12288.00 | root         |               | MppVersion: 1, data:ExchangeSender_11                 |
+| └─ExchangeSender_11     | 12288.00 | mpp[tiflash] |               | ExchangeType: PassThrough                             |
+|   └─TableFullScan_9     | 12288.00 | mpp[tiflash] | table:t1      | pushed down filter:lt(test.t1.a, 1), keep order:false |
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+```
+
+过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:36]:
```suggestion
在此示例中，过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
```

File: tiflash/tiflash-late-materialization.md [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+
+> **警告：**
+>
+> 该功能目前是实验性功能，其形式和使用方法可能会在未来版本中发生变化。
+
+本文档介绍通过使用 TiFlash 延迟物化的方式来加速 Online Analytical Processing (OLAP) 场景的查询。
+
+默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件对数据进行过滤、聚合等计算任务。延迟物化是一种优化方式，它支持下推部分过滤条件到 TableScan 算子，即先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
+
+如果希望提升 OLAP 场景部分查询的性能，可以在 session 级别或 global 级别开启 TiFlash 延迟物化功能。你可以通过修改变量 `tidb_opt_enable_late_materialization` 的值来选择是否启用 TiFlash 延迟物化功能。
```
Comment (hfxsd) [inline tiflash/tiflash-late-materialization.md:16]: `tidb_opt_enable_late_materialization` 加上链接

File: tiflash/tiflash-late-materialization.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+| └─ExchangeSender_11     | 12288.00 | mpp[tiflash] |               | ExchangeType: PassThrough                             |
+|   └─TableFullScan_9     | 12288.00 | mpp[tiflash] | table:t1      | pushed down filter:lt(test.t1.a, 1), keep order:false |
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+```
+
+在此示例中，过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
+
+## 启用和禁用 TiFlash 延迟物化
+
+默认情况下，session 和 global 级别的变量 `tidb_opt_enable_late_materialization=OFF`，即没有开启 TiFlash 延迟物化功能。你可以通过以下语句来查看对应的变量信息。
```
Comment (hfxsd) [inline tiflash/tiflash-late-materialization.md:40]:
```suggestion
默认情况下，session 和 global 级别的变量 `tidb_opt_enable_late_materialization=OFF`，即禁用 TiFlash 延迟物化功能。你可以通过以下语句来查看对应的变量信息。
```

File: tiflash/tiflash-late-materialization.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+
+- 如果需要在当前 session 中启用 TiFlash 延迟物化功能，可以通过以下语句设置:
+    ```
+    set session tidb_opt_enable_late_materialization=ON;
+    ```
+- 如果需要在 global 级别启用 TiFlash 延迟物化功能，可以通过以下语句设置：
+    ```
+    set global tidb_opt_enable_late_materialization=ON;
+    ```
+    设置后，新建的会话中 session 和 global 级别 `tidb_opt_enable_late_materialization` 都将默认启用新值。
```
Comment (hfxsd) [inline tiflash/tiflash-late-materialization.md:76]:
```suggestion
- 如果需要在当前 session 中启用 TiFlash 延迟物化功能，可以通过以下语句设置:
    
    ```
    set session tidb_opt_enable_late_materialization=ON;
    ```

- 如果需要在 global 级别启用 TiFlash 延迟物化功能，可以通过以下语句设置：

    ```
    set global tidb_opt_enable_late_materialization=ON;
    ```

    设置后，新建的会话中 session 和 global 级别 `tidb_opt_enable_late_materialization` 都将开启延迟物化功能。
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: tiflash/tiflash-late-materialization.md (hunk 19 above)
Comment (Lloyd-Pottiger) [reply tiflash/tiflash-late-materialization.md:40]: 目前默认不开始，所以我觉得 `没有开启` 好一点

Comment (qiancai) [reply tiflash/tiflash-late-materialization.md:40]: 要不改成 “即未开启“？

File: tiflash/tiflash-late-materialization.md [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+| tidb_opt_enable_late_materialization | OFF   |
++--------------------------------------+-------+
+```
+
+变量 `tidb_opt_enable_late_materialization` 支持 session 级别和 global 级别的修改。
+
+- 如果需要在当前 session 中启用 TiFlash 延迟物化功能，可以通过以下语句设置:
+    
+    ```sql
+    set session tidb_opt_enable_late_materialization=ON;
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:71]:
```suggestion
    SET SESSION tidb_opt_enable_late_materialization=ON;
```

File: tiflash/tiflash-late-materialization.md [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+    ```sql
+    set global tidb_opt_enable_late_materialization=ON;
+    ```
+    
+    设置后，新建的会话中 session 和 global 级别 `tidb_opt_enable_late_materialization` 都将默认启用新值。
+
+如需禁用 TiFlash 延迟物化功能，可以通过以下语句设置：
+
+```sql
+set session tidb_opt_enable_late_materialization=OFF;
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:85]:
```suggestion
SET SESSION tidb_opt_enable_late_materialization=OFF;
```

File: tiflash/tiflash-late-materialization.md [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+```sql
+set global tidb_opt_enable_late_materialization=OFF;
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:89]:
```suggestion
SET GLOBAL tidb_opt_enable_late_materialization=OFF;
```

File: tiflash/tiflash-late-materialization.md [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+- 如果需要在当前 session 中启用 TiFlash 延迟物化功能，可以通过以下语句设置:
+    
+    ```sql
+    set session tidb_opt_enable_late_materialization=ON;
+    ```
+
+- 如果需要在 global 级别启用 TiFlash 延迟物化功能，可以通过以下语句设置：
+    
+    ```sql
+    set global tidb_opt_enable_late_materialization=ON;
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:77]:
```suggestion
    SET GLOBAL tidb_opt_enable_late_materialization=ON;
```

File: tiflash/tiflash-late-materialization.md [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+| └─ExchangeSender_11     | 12288.00 | mpp[tiflash] |               | ExchangeType: PassThrough                             |
+|   └─TableFullScan_9     | 12288.00 | mpp[tiflash] | table:t1      | pushed down filter:lt(test.t1.a, 1), keep order:false |
++-------------------------+----------+--------------+---------------+-------------------------------------------------------+
+```
+
+在此示例中，过滤条件 `a < 1` 被下推到了 TableScan 算子，TiFlash 会先读取列 `a` 的全部数据，然后根据过滤条件 `a < 1` 进行过滤，得到符合条件的行后，再读取这些行的列 `b` 和 `c` 的数据。
+
+## 启用和禁用 TiFlash 延迟物化
+
+默认情况下，session 和 global 级别的变量 `tidb_opt_enable_late_materialization=OFF`，即没有开启 TiFlash 延迟物化功能。你可以通过以下语句来查看对应的变量信息。
```
Comment (Lloyd-Pottiger) [inline tiflash/tiflash-late-materialization.md:40]:
```suggestion
默认情况下，session 和 global 级别的变量 `tidb_opt_enable_late_materialization=OFF`，即未开启 TiFlash 延迟物化功能。你可以通过以下语句来查看对应的变量信息。
```

File: tiflash/tiflash-late-materialization.md (hunk 19 above)
Comment (Lloyd-Pottiger) [reply tiflash/tiflash-late-materialization.md]: LGTM，改了

**Review by qiancai — APPROVED** [review]

**Comment by Lloyd-Pottiger** [comment]
（不合嘛？

**Comment by qiancai** [comment]
> （不合嘛？

处理英文的时候，可能还会改中文，等英文好了后，一起再合吧

File: system-variables.md [hunk 26]
```diff
@@ -1379,6 +1379,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 - 该变量为 TiDB 内部变量，**不推荐使用**，否则可能会造成数据正确性问题。
 
+### `tidb_opt_enable_late_materialization` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 表示是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。
+- 默认情况下，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 可以先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
```
Comment (qiancai) [inline system-variables.md:1389]:
```suggestion
- 默认情况下，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 可以先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
```

File: tiflash/tiflash-late-materialization.md (hunk 21 above)
Comment (qiancai) [reply tiflash/tiflash-late-materialization.md:71]: SQL 关键字需要尽量大写，符合规范， SET SESSION 参考 
- https://github.com/pingcap/docs/issues/699
-  https://docs.pingcap.com/tidb/dev/sql-statement-set-variable 
- https://dev.mysql.com/doc/refman/8.0/en/set-variable.html#:~:text=To%20assign%20a%20value%20to%20a%20session%20system%20variable%2C%20precede,TRADITIONAL'%3B%20SET%20%40%40SESSION.

File: tiflash/tiflash-late-materialization.md [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+```sql
+set global tidb_opt_enable_late_materialization=OFF;
+```
+
+## 实现机制
+
+当有过滤条件下推到 TableScan 算子时，TableScan 算子的执行过程主要包括了以下步骤：
+
+1. 读取 `<handle, del_mark, version>` 三列，执行多版本并发控制 (MVCC) 过滤，生成 MVCC Bitmap。
+2. 读取过滤条件相关的列，执行过滤条件，生成 Filter Bitmap。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:97]:
```suggestion
2. 读取过滤条件相关的列，过滤满足条件的行，生成 Filter Bitmap。
```

File: tiflash/tiflash-late-materialization.md [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+>
+> 该功能目前是实验性功能，其形式和使用方法可能会在未来版本中发生变化。
+
+本文档介绍通过使用 TiFlash 延迟物化的方式来加速 Online Analytical Processing (OLAP) 场景的查询。
+
+默认情况下，TiFlash 会先读取查询所需列的全部数据，然后再根据查询条件对数据进行过滤、聚合等计算任务。延迟物化是一种优化方式，它支持下推部分过滤条件到 TableScan 算子，即先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少扫描 IO 和数据解析的计算量。
+
+如果希望提升 OLAP 场景部分查询的性能，可以在 session 级别或 global 级别开启 TiFlash 延迟物化功能。你可以通过修改变量 [`tidb_opt_enable_late_materialization`](/system-variables.md#tidb_opt_enable_late_materialization-从-v700-版本开始引入) 的值来选择是否启用 TiFlash 延迟物化功能。
+
+启用 TiFlash 延迟物化功能后，TiDB 优化器会根据统计信息和查询的过滤条件，决定哪些过滤条件会被下推。优化器会优先考虑下推过滤率高的过滤条件，详细算法可以参考 [RFC 文档](https://github.com/pingcap/tidb/tree/master/docs/design/2022-12-06-support-late-materialization.md)。
```
Comment (qiancai) [inline tiflash/tiflash-late-materialization.md:18]: "下推过滤率高"，英文对应 “low selectivity”？

File: tiflash/tiflash-late-materialization.md [hunk 29]
```diff
@@ -0,0 +1,100 @@
...
+
+## 实现机制
+
+当有过滤条件下推到 TableScan 算子时，TableScan 算子的执行过程主要包括了以下步骤：
+
+1. 读取 `<handle, del_mark, version>` 三列，执行多版本并发控制 (MVCC) 过滤，生成 MVCC Bitmap。
+2. 读取过滤条件相关的列，过滤满足条件的行，生成 Filter Bitmap。
+3. 将 MVCC Bitmap 和 Filter Bitmap 进行与操作 (AND)，生成 Final Bitmap。
+4. 根据 Final Bitmap 读取剩余列的对应的行。
+5. 合并两次读取的数据，返回结果。
```
Comment (Lloyd-Pottiger) [inline tiflash/tiflash-late-materialization.md:100]:
```suggestion
5. 合并第 2 步和第 4 步中读取的数据，返回结果。
```

File: system-variables.md [hunk 30]
```diff
@@ -2579,6 +2579,15 @@ mysql> desc select count(distinct a) from test.t;
 - 默认值：`ON`
 - 这个变量用来控制优化器是否开启交叉估算。
 
+### `tidb_opt_enable_late_materialization` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 表示是否启用 [TiFlash 延迟物化](/tiflash/tiflash-late-materialization.md)功能。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply system-variables.md:2588]: reason: use the same style as the descriptions of other variables

**Pull request merged**